cmd/drone-docker-ecr/main.go

```diff
@@ -1,5 +1,113 @@
 package main
 
+import (
+	"encoding/base64"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"os"
+	"os/exec"
+)
+
+const (
+	DOCKER_USER    = "AWS"
+	DEFAULT_REGION = "us-east-1"
+)
+
 func main() {
+	var registryIds []*string
+
+	ecrRegion := DEFAULT_REGION
+
+	if getenv("ECR_REGION") != "" {
+		os.Setenv("PLUGIN_REGION", getenv("ECR_REGION"))
+	}
+
+	if accessKey := getenv("ECR_ACCESS_KEY"); accessKey != "" {
```

Comment:
You should call getenv with all possible env names, not one by one, look at the gcr command